### ISSUE TITLE
Debugger runs past all breakpoiints and exits when expand object in variable view

### ISSUE THREAD

**Issue by jabamaus** [rubyide/vscode-ruby#754]
### Your environment

- `vscode-ruby` version: 0.28.1
- Ruby version: 3.0.2
- Ruby version manager (if any): none
- VS Code version: 1.58.2
- Operating System: windows 10
- Using language server? (eg `useLanguageServer` is true in your configuration?) No

### Expected behavior

I should be able to expand the object in the watch window and view its variables when I'm on a breakpoint.

### Actual behavior

The variable view briefly expands to show the member variables of the objects and then after a split second goes blank. Then the debugger exits without hitting any other breakpoints even though some are set. Its as if expanding the variable view is crashing the debugger. If my code is giving some kind of bad value or contains a bug there should be an error in the variable view. It shouldn't just go blank. I've seen this happen many times but its intermittent. Its driving me crazy. I think it happens more when my own code is in a broken state but if I put the breakpoint before the broken code it still does it.

Any help appreciated! This is the one and only issue with the debugger as far as I'm concerned.

**Comment by pascalbetz** [comment]
I have a similar issue where i can see the breakpoints added to the code

```
some-app_1  | 1: <breakpointAdded no="3" location="...../some-app/app/controllers/hello_worlds_controller.rb:4"/>
```

but is never hit. Any hints how I can debug this would be appreciated.

**Comment by jabamaus** [comment]
This drives me mad and has been happening with my software for the last three years. One tell tale sign is that it only happens when I expand my top level object in the variable view. My top level objects ends up referencing everything in the system and everything holds back references to it. This bug makes it impossible to debug many of my issues and is so frustrating. It really feels like something needs wrapping in an exception handler. I have already spent ages trying to figure this out but I will try again. I don't even know if the problem is in debase, ruby-debug-ide or the ruby vscode plugin...

**Comment by pascalbetz** [comment]
I don't even get anything in the variable view. Breakpoint is never hit. I'll try and reproduce whit an easier example as my setup includes an existing app that runs in docker. so i'll start with a local ruby script and see if my setup is correct.

**Comment by pascalbetz** [comment]
With some help from Stackoverflow i figured out the issue: i had an error in my VS Code launch config: the remoteWorkspaceRoot was set to use the local and not the remote path.

**Comment by jabamaus** [comment]
Glad you solved it :) I finally solved my issue too. It was a really evil one. My system uses BasicObjects combined with method_missing, and when the variable view was trying display a member variable that was one of these objects it was causing new objects in my code that were being created as a consequence of being debugged. To prevent this I had to implement a dummy nil? method (which happens to be the first method ruby_debug_ide calls when figuring out how to print an object).

I added this to my BasicObject subclass to fix the problem. Now the whole debugger doesn't go down whenever I inspect my main object. Finally!

    if ::JABA.ruby_debug_ide?
      def to_s
        'BasicObject'
      end
      
      def nil?
        ::Kernel.raise to_s
      end
    end

By the way, I wanted to fix this at ruby_debug_ide level but I couldn't find a way of asking an object if it was a BasicObject without it going through to my method_missing.
